robolectric/src/main/java/org/robolectric/internal/dependency/LocalDependencyResolver.java

```diff
@@ -8,9 +8,11 @@
 
 public class LocalDependencyResolver implements DependencyResolver {
   private File offlineJarDir;
+  private final DependencyResolver dependencyResolver;
 
-  public LocalDependencyResolver(File offlineJarDir) {
+  public LocalDependencyResolver(DependencyResolver dependencyResolver, File offlineJarDir) {
```

Comment:
I don't think this needs to change. CachedDependencyResolver just decorates any other single DependencyResolver doesn't it? I can imagine that you would just override this method:-

  protected DependencyResolver getJarResolver() {
    return new CompositeDependencyResolver(
       new LocalDependencyResolver(new File("/path/to/my/local/repo")).
       new CachedDependencyResolver(
           new MavenDependencyResolver(),
           new File("/path/to/my/cache"),
           CACHE_EXPIREY_TIME))
    }
  }

I think the idea of the CachedDependencyResolver was just to support the offline case where maven was not available so developers can still work with the cached jars. Obviously you don't need to cache results from LocalDependencyResolver.